opensearch-transport/opensearch-transport.gemspec

```diff
@@ -84,6 +87,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'simplecov'
   s.add_development_dependency 'test-unit', '~> 2'
   s.add_development_dependency 'typhoeus', '~> 1.4'
+  s.add_development_dependency 'faraday-typhoeus' unless RUBY_VERSION.to_f < 2.7
```

Comment:
I think you do have to fix this one or we may have some interesting problems :(

```
> "1.5.8".to_f > "1.5.7".to_f
  => false
```

Reply:
Nope, because being less than 2.7.0 is the constraint we want. The other point values don't matter.

Reply:
I see, it just happens that `'1.2.3'.to_f` is 1.2, but I think we shouldn't rely on it since it's semantically incorrect. I got burned many times. It's too easy for another developer to make a mistake. I'd prefer this was changed, please. 

https://code.dblock.org/2020/07/16/comparing-version-numbers-in-ruby.html

Reply:
I switched to Gem::Version comparison, after I made sure that the minimum Ruby version supported it without explicitly calling `require 'rubygems'` to get the Gem stuff.